File: java/flowcamel-core/src/main/java/com/flowcamel/core/model/FlowDefinition.java
package com.flowcamel.core.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.ArrayList;
import java.util.List;

@JsonIgnoreProperties(ignoreUnknown = true)
public class FlowDefinition {
  public String id = "";
  public String name = "";
  public String routeId = "";
  public List<FlowNode> nodes = new ArrayList<>();
  public List<FlowEdge> edges = new ArrayList<>();
}

File: packages/core/src/model/FlowDefinition.ts
import type { FlowEdge } from './FlowEdge.js';
import type { FlowNode } from './FlowNode.js';

/** One Camel route: nodes/edges on the canvas tab. */
export interface FlowDefinition {
  id: string;
  name: string;
  routeId: string;
  nodes: FlowNode[];
  edges: FlowEdge[];
}
